### PR TITLE
This is a partial cleanup of the remaining filesystem loose ends.

### PR DESCRIPTION
**Pull request by bkentel**
Updated `game::move_save_to_graveyard()`.

### CODE REVIEW

**Comment by bkentel** [comment]
Might be a good idea to add back a couple of the comments (thoughts?).

This partially addresses the TODO in the initial change over to filsystem.* .

File: src/game.cpp [hunk 1]
```diff
@@ -4247,38 +4247,35 @@ void game::death_screen()
...
-        const std::string srcpath = savedir + "/" + name;
-        // this might fail if the graveyard dir does not exist
-        if( !rename_file( srcpath, dstpath ) ) {
-            // Permadeath! The player file must be gone!
-            if( !remove_file( srcpath ) ) {
-                // AHHH, who or what prevents permadeath?
-                debugmsg( "could not remove file %s", srcpath.c_str() );
-            }
+
+        debugmsg("could not rename file '%s' to '%s'", src_path.c_str(), dst_path.substr());
```
Comment (kevingranade) [inline src/game.cpp]: Jenkins is complaining about this, I think it wants dst_path.c_str(), not substr.  Why substr anyway?

**Comment by kevingranade** [comment]
Looks reasonable, but I need signoff from people building on the windows platforms.

File: src/game.cpp (hunk 1 above)
Comment (bkentel) [inline src/game.cpp]: auto-complete induced type; I was updating the errors messages just before that last commit after testing the actual implementation.

**Comment by KA101** [comment]
OK, compiled fine and a quick suicide moved the immediate map to the graveyard.  Will make a full save and test again.

**Comment by bkentel** [comment]
@KA101 Thanks for taking the time to test.

**Comment by KA101** [comment]
Looks working on my end.  Will grab this if Kevin doesn't.

**Pull request merged**